force-app/main/default/triggers/LeadTrigger.trigger

```diff
@@ -15,21 +15,22 @@
  * - It's essential to test the trigger thoroughly after making any changes to ensure its correct functionality.
  * - Debugging skills will be tested, so students should look out for discrepancies between the expected and actual behavior.
  */
-trigger LeadTrigger on Lead(before insert) {
-	switch on Trigger.operationType {
-		when BEFORE_INSERT {
-			LeadTriggerHandler.handleTitleNormalization(Trigger.new);
-			LeadTriggerHandler.handleAutoLeadScoring(Trigger.new);
-		}
-		when BEFORE_UPDATE {
-			LeadTriggerHandler.handleTitleNormalization(Trigger.new);
-			LeadTriggerHandler.handleAutoLeadScoring(Trigger.new);
-		}
-		when AFTER_INSERT {
-			LeadTriggerHandler.handleLeadAutoConvert(Trigger.new);
-		}
-		when AFTER_UPDATE {
-			LeadTriggerHandler.handleLeadAutoConvert(Trigger.new);
+trigger LeadTrigger on Lead(before insert, before update, after insert, after update) {
+	LeadTriggerHandler handler = new LeadTriggerHandler(Trigger.isExecuting, Trigger.size);
```

Comment:
For practice make the lead methods static so that you don't need to instantiate a handler